src/python/pants/backend/python/providers/pyenv/rules.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent  # noqa: PNT20
+
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules.pex import PythonProvider
+from pants.backend.python.util_rules.pex import rules as pex_rules
+from pants.backend.python.util_rules.pex_environment import PythonExecutable
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolRequest,
+    TemplatedExternalTool,
+)
+from pants.core.util_rules.external_tool import rules as external_tools_rules
+from pants.engine.env_vars import EnvironmentVars, EnvironmentVarsRequest
+from pants.engine.fs import CreateDigest, FileContent
+from pants.engine.internals.native_engine import Digest, MergeDigests
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import StrListOption
+from pants.util.frozendict import FrozenDict
+from pants.util.meta import classproperty
+
+
+class PyenvSubsystem(TemplatedExternalTool):
+    options_scope = "pyenv"
+    name = "pyenv"
+    help = "pyenv (https://github.com/pyenv/pyenv)."
+
+    default_version = "2.3.13"
+    default_url_template = "https://github.com/pyenv/pyenv/archive/refs/tags/v{version}.tar.gz"
+
+    python_configure_opts = StrListOption(
+        help="Flags to use when configuring CPython.",
+        advanced=True,
+    )
```

Comment:
This doesn't invalidate the existing Python. That's a choice I made. Should it though?

Reply:
Probably? Also, is this actually limited to cpython? Seems like I could specify other pythons in my ICs?

Reply:
This specific flag is limited to CPython: https://github.com/pyenv/pyenv/blob/master/plugins/python-build/README.md#special-environment-variables

If we wanted it to invalidate, we'd need to make the path have a hash of this data.

Reply:
I removed this option. `--enable-optimizations` is SUPER slow.